src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java

```diff
@@ -0,0 +1,225 @@
+     */
+    private void positionShip(final String shipType, final Integer shipQuantity, final ShipFactory shipFactory) {
+        Random random = new Random();
+        for (int i = 0; i < shipQuantity; i++) {
+            boolean placeShip = false;
+            while (!placeShip) {
+                int row = random.nextInt(ships.length - 1);
+                int column = random.nextInt(ships.length - 1);
+                boolean horizontal = random.nextBoolean();
+                Ship newShip = shipFactory.createShip(shipType);
+                if (newShip.okToPlaceShipAt(row, column, horizontal, this)) {
+                    newShip.placeShipAt(row, column, horizontal, this);
+                    placeShip = true;
+                }
+            }
+        }
+    }
+
+    /**
+     * Method that verify if a position is occupied.
+     *
+     * @param row    row position.
+     * @param column column position.
+     * @return True if the given position is occupied, false otherwise.
+     */
+    boolean isOccupied(int row, int column) {
+        return !(ships[row][column] instanceof EmptySea);
+    }
+
+    /**
+     * Method that shoot in a position.
+     *
+     * @param row    row position.
+     * @param column column position.
+     * @return True if the shoot was accurate, False otherwise.
+     */
+    boolean shootAt(int row, int column) {
+        shotsFired++;
+        if (isOccupied(row, column) && !ships[row][column].isSunk()) {
+            hitCount++;
+            if (ships[row][column].shootAt(row, column)) {
+                if (ships[row][column].isSunk()) {
+                    shipsSunk++;
+                }
+                return true;
+            }
+
+            return false;
+        }
+        ships[row][column].shootAt(row, column);
+        return false;
+    }
+
+    /**
+     * Method that print the ocean.
+     */
+    public void print() {
+        printColumnPosition();
+        for (int i = 0; i < ships.length; i++) {
+            System.out.println("");
+            System.out.print(i);
+            for (int j = 0; j < ships.length; j++) {
+                if (ships[i][j].toString().equals("S") && !ships[i][j].hit[ships[i][j].getHitIndex(i, j)]) {
+                    System.out.print(" . ");
+                } else {
+                    System.out.print(" " + ships[i][j] + " ");
+                }
+            }
+            System.out.println("");
+        }
+        System.out.println("");
+    }
+
+    /**
+     * Method tha print column numbers in the ocean.
+     */
+    private void printColumnPosition() {
+        final int maxNumberAddCero = 10;
+        StringBuilder arrayPosition = new StringBuilder(" ");
+        for (int i = 0; i < ships.length; i++) {
+            if (i < maxNumberAddCero) {
+                arrayPosition.append("0");
+            }
+            arrayPosition.append(i);
+            arrayPosition.append(" ");
+
+        }
+        System.out.print(arrayPosition);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)